File: competition/Feb 21. 2021/Solution5686.cpp
#include<iostream>
#include<vector>

#include<unordered_set>

using namespace std;

class Solution5686 {
public:
    vector<int> minOperations(string boxes) {
        unordered_set<int> set;
        for(int i = 0; i < boxes.length(); i++){
            if(boxes[i] == '1'){
                set.insert(i);
            }
        }
        vector<int> result(boxes.length());
        for(int i = 0; i < boxes.length(); i++){
            int cur = 0;
            for(auto ele: set){
                cur += abs(ele - i);
            }
            result[i] = cur;
        }
        return result;
    }
};

File: competition/Feb 21. 2021/Solution5686.java
import java.util.*;

class Solution5686 {
    public int[] minOperations(String boxes) {
    	Set<Integer> set = new HashSet<>();
    	for(int i = 0; i < boxes.length(); i++){
    		if(boxes.charAt(i) == '1'){
    			set.add(i);
    		}
    	}
    	int[] result = new int[boxes.length()];
    	for(int i = 0; i < boxes.length(); i++){
    		int cur = 0;
    		for(Integer ele: set){
    			cur += Math.abs(ele - i);
    		}
    		result[i] = cur;
    	}
    	return result;
    }
}